src/librustdoc/passes/html_tags.rs

```diff
@@ -94,6 +94,14 @@ fn extract_path_backwards(text: &str, end_pos: usize) -> Option<usize> {
     if current_pos == end_pos { None } else { Some(current_pos) }
 }
 
+fn is_valid_for_html_tag_name(c: char, is_empty: bool) -> bool {
+    // https://spec.commonmark.org/0.30/#raw-html
+    //
+    // > A tag name consists of an ASCII letter followed by zero or more ASCII letters, digits, or
+    // > hyphens (-).
+    c.is_ascii_alphabetic() || (c.is_ascii_digit() && !is_empty) || (c == '-' && !is_empty)
```

Comment:
Completely anecdotical but maybe like this would look better:

```suggestion
    c.is_ascii_alphabetic() || !is_empty && (c == '-' || c.is_ascii_digit())
```

?

Less comparisons and maybe a bit more clear. But again: completely anecdotical.